pkg/mover/auth.go

```diff
@@ -25,16 +31,41 @@ func (repo ContainerRepository) Resolve(resource authn.Resource) (authn.Authenti
 // See https://pkg.go.dev/github.com/google/go-containerregistry/pkg/authn#Authenticator
 //
 // Returns an authn.AuthConfig with a user / password pair to be used for authentication
-func (repo ContainerRepository) Authorization() (*authn.AuthConfig, error) {
+func (repo *OCICredentials) Authorization() (*authn.AuthConfig, error) {
 	return &authn.AuthConfig{Username: repo.Username, Password: repo.Password}, nil
 }
 
-// Define a container images keychain based on the settings provided via the containers struct
-// If useDefaultKeychain is set, use config/docker.json otherwise load the provided creds (if any)
-func getContainersKeychain(c Containers) authn.Keychain {
+// Define a container registry keychain based on the settings provided in containers Auth
+// If useDefaultKeychain is set, use config/docker.json otherwise it will load the provided credentials (if any)
+func getContainersKeychain(c *ContainersAuth) (authn.Keychain, error) {
+	// No credentials provided
+	if !c.UseDefaultLocalKeychain && c.Credentials == nil {
+		return nil, errors.New("either local keychain or explicit credentials are required")
+	}
+
+	if c.UseDefaultLocalKeychain && c.Credentials != nil {
+		return nil, errors.New("you can use either local keychain or explicit credentials not both")
+	}
+
 	if c.UseDefaultLocalKeychain {
-		return authn.DefaultKeychain
+		return authn.DefaultKeychain, nil
+	}
+
+	return validateOCICredentials(c.Credentials)
+}
+
+// validate if the provided OCI credentials are valid
+// They include a username, password and a valid (RFC 3986 URI authority) serverName
+func validateOCICredentials(c *OCICredentials) (authn.Keychain, error) {
+	if c.Username == "" || c.Password == "" || c.Server == "" {
+		return nil, errors.New("OCI credentials require an username, password and a server name")
+	}
+
+	// See https://github.com/google/go-containerregistry/blob/main/pkg/name/registry.go#L97
+	// Valid RFC 3986 URI authority
+	if uri, err := url.Parse("//" + c.Server); err != nil || uri.Host != c.Server {
+		return nil, fmt.Errorf("credentials server name %q must not contain a scheme (\"http://\") nor a path. Example of valid registry names are \"myregistry.io\" or \"myregistry.io:9999\"", c.Server)
```

Comment:
I guess this error message does not apply if `uri.Host != c.Server`

Reply:
I think it applies "...nor a path"

Reply:
right!